File: lib/api/jobs.ts
/**
 * SatQuery AI — Jobs API Client
 */

import { apiClient } from './client';
import type { AnalysisStatusType } from './analysis';

export interface JobStatusDTO {
  job_id: string;
  status: AnalysisStatusType;
  progress: number;
  message?: string | null;
  result_id?: string | null;
}

export async function getJob(jobId: string): Promise<JobStatusDTO> {
  return apiClient<JobStatusDTO>(`/api/jobs/${jobId}`);
}

File: backend/app/api/routes/jobs.py
from typing import Optional
from fastapi import APIRouter, HTTPException, status
from pydantic import BaseModel
from app.domain.tasks import AnalysisStatus

router = APIRouter(prefix="/api/jobs", tags=["Jobs"])


class JobStatusResponse(BaseModel):
    job_id: str
    status: AnalysisStatus
    progress: float
    message: Optional[str] = None
    result_id: Optional[str] = None


@router.get("/{job_id}", response_model=JobStatusResponse)
async def get_job_status(job_id: str) -> JobStatusResponse:
    """
    Check asynchronous job execution status.
    In this foundation stage, any queried job returns a clean state or 404.
    """
    raise HTTPException(
        status_code=status.HTTP_404_NOT_FOUND,
        detail=f"Job '{job_id}' not found. Asynchronous queuing pipeline will be wired in model integration."
    )
